src/s1reader/s1_reader.py

```diff
@@ -672,15 +672,14 @@ def burst_from_xml(annotation_path: str, orbit_path: str, tiff_path: str,
 
         rfi_annotation_path = annotation_path.replace('annotation/', 'annotation/rfi/rfi-')
         # Load RFI information if available
-        if os.path.exists(rfi_annotation_path):
+        try:
             with open_method(rfi_annotation_path, 'r') as f_rads:
                 tree_rads = ET.parse(f_rads)
                 burst_rfi_info_swath = SwathRfiInfo.from_et(tree_rads,
                                                             tree_lads,
                                                             ipf_version)
 
-        else:
-            warnings.warn(f"No RFI metadata found at {rfi_annotation_path}.")
+        except (FileNotFoundError, KeyError):
             burst_rfi_info_swath = None
```

Comment:
```suggestion
        except (FileNotFoundError, KeyError):
            if ipf_version >= RFI_INFO_AVAILABLE_FROM:
                warnings.warn(f"RFI annotation expected (IPF version={ipf_version}"
                              f" >= {RFI_INFO_AVAILABLE_FROM}), but not loaded.")
            burst_rfi_info_swath = None
```

I think it's still be helpful to display warning message to early notice the bug.
The constant `RFI_INFO_AVAILABLE_FROM` is defined in `s1_annotation.py`. You might need to import that from there.